syntax= "proto3";
package socialnetwork;

service SocialNetwork {
  rpc follow (followRequest) returns (reqResponse);
  rpc unfollow (followRequest) returns (reqResponse);
  rpc list (user) returns (followersList);
  rpc register(user) returns (reqResponse);
}

message reqResponse{
    bool success = 1;
}

message followersList {
    repeated string followers = 1;
}

message user {
    string userId = 1;
}

message followRequest {
  string followee = 1;
  string follower = 2;
}

message status {
    string reply = 1;
}
